Simplenote/PasskeyAuthenticator.swift

```diff
@@ -3,126 +3,96 @@ import AuthenticationServices
+        case .registrationFailed:
+            return NSLocalizedString("Registration Failed", comment: "Error message that passkey registration failed")
+        }
+    }
 }
 
-@objcMembers
-class PasskeyAuthenticator: NSObject {
-    typealias PresentationContext = ASAuthorizationControllerPresentationContextProviding
-    let authenticator: SPAuthenticator
-    let passkeyRemote = PasskeyRemote()
-
-    var registrationEmail: String?
+class PasskeyAuthControllerDelegate: NSObject, ASAuthorizationControllerDelegate {
 
-    @objc
-    init(authenticator: SPAuthenticator) {
-        self.authenticator = authenticator
-    }
+    var onCompletion: ((Result<PasskeyAuthResponse, Error>) -> Void)?
 
-    // MARK: - Registration
-    //
-    func registerPasskey(for email: String, password: String, in presentationContext: PresentationContext) async throws {
-        registrationEmail = email
-        do {
-            guard let data = try await passkeyRemote.requestChallengeResponseToCreatePasskey(forEmail: email, password: password) else {
-                throw PasskeyError.couldNotRequestRegistrationChallenge
-            }
-            let passkeyChallenge = try JSONDecoder().decode(PasskeyRegistrationChallenge.self, from: data)
-            attemptRegistration(with: passkeyChallenge, presentationContext: presentationContext)
-        } catch {
-            throw PasskeyError.couldNotRequestRegistrationChallenge
-        }
+    public func authorizationController(controller: ASAuthorizationController, didCompleteWithError error: any Error) {
+        onCompletion?(.failure(error))
     }
 
-    private func attemptRegistration(with passkeyChallenge: PasskeyRegistrationChallenge, presentationContext: PresentationContext) {
-        guard let challengeData = passkeyChallenge.challengeData,
-              let userID = passkeyChallenge.userID else {
+    public func authorizationController(controller: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
+        guard let credential = authorization.credential as? ASAuthorizationPlatformPublicKeyCredentialAssertion else {
+            onCompletion?(.failure(PasskeyError.authFailed))
             return
         }
 
-        let platformProvider = ASAuthorizationPlatformPublicKeyCredentialProvider(relyingPartyIdentifier: passkeyChallenge.relayingPartyIdentifier)
-        let platformKeyRequest = platformProvider.createCredentialRegistrationRequest(challenge: challengeData, name: passkeyChallenge.displayName, userID: userID)
-        let authController = ASAuthorizationController(authorizationRequests: [platformKeyRequest])
-        authController.delegate = self
-        authController.presentationContextProvider = presentationContext
-        authController.performRequests()
+        let response = PasskeyAuthResponse(from: credential)
+        onCompletion?(.success(response))
```

Comment:
Same note as in `PasskeyRegistrator`, there's a retain cycle between this class and `PasskeyAuthenticator`.

Can be fixed exactly the same way, either clearing `onCompletion` after invoking it, or making `PasskeyAuthenticator` responsible for doing so.